src/utils/download_file.ts

```diff
@@ -34,19 +39,8 @@ export async function downloadFile(
 			downloadedLength += chunk.length;
 			downloadProgressCallback && downloadProgressCallback((downloadedLength / totalLength) * 100);
 		});
-		return new Promise((resolve) => {
-			data.pipe(writer);
-			let error: Error | null = null;
-			writer.on('error', (err) => {
-				error = err;
-				writer.close();
-			});
-			writer.on('close', () => {
-				if (!error) {
-					resolve({ pathToFile, contentType });
-				}
-			});
-		});
+		await pipeline(data, writer);
```

Comment:
What behavior do you get right now if an error is thrown from one of the streams? e.g. if you pass a dead URL to try and download? Does pipeline throw?

Reply:
I only ask because we have no error handler here so I'm just wondering if the error ends up getting thrown (and whether that's what we want here).